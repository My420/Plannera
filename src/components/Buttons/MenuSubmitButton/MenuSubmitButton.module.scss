@import '../../../sass/variables.scss';

.button {
  opacity: 1;
  box-sizing: border-box;

  width: 300px;
  height: 40px;

  border: none;
  border-radius: 3px;

  background-color: $bg-app-color;

  font-weight: 700;
  color: $fonts-auth-color;

  transition: all 500ms linear;
}

.button:disabled {
  opacity: 0;
}
