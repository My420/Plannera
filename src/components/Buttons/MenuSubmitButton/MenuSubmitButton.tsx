/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import styles from './MenuSubmitButton.module.scss';

/* React.HTMLAttributes<HTMLButtonElement> disabled and other attr don't work */
export interface MenuSubmitButtonProps
  extends React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
  > {
  text: string;
  name: string;
  onButtonClick: (evt: React.MouseEvent<HTMLButtonElement>) => void;
  extensionClass?: string;
}

const MenuSubmitButton: React.FC<MenuSubmitButtonProps> = ({
  text,
  name,
  extensionClass,
  onButtonClick,
  ...rest
}) => (
  <button
    className={`${styles.button} ${extensionClass}`}
    type="submit"
    name={name}
    onClick={onButtonClick}
    {...rest}
  >
    {text}
  </button>
);

MenuSubmitButton.defaultProps = {
  extensionClass: '',
};
export default MenuSubmitButton;
