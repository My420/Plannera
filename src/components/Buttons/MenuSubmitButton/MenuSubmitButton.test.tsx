import React from 'react';
import { shallow } from 'enzyme';
import MenuSubmitButton from './MenuSubmitButton';

describe('Test MenuSubmitButton component', () => {
  const text = 'Text Button';
  const name = 'testBtn';
  const testAttr = 'test';
  const mockFn = jest.fn();
  const wrapper = shallow(
    <MenuSubmitButton text={text} name={name} onButtonClick={mockFn} data-test={testAttr} />,
  );

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render  tag button', () => {
    const tag = wrapper.find('button');
    expect(tag).toHaveLength(1);
  });

  test('tag button should have correct default className', () => {
    const actual = wrapper.find('button').prop('className');
    const expected = 'button ';
    expect(actual).toBe(expected);
  });

  test('tag button should have correct attr type', () => {
    const actual = wrapper.find('button').prop('type');
    const expected = 'submit';
    expect(actual).toBe(expected);
  });

  test('tag button should have correct attr name', () => {
    const actual = wrapper.find('button').prop('name');
    const expected = name;
    expect(actual).toBe(expected);
  });

  test('tag button should have correct attr onClick', () => {
    const actual = wrapper.find('button').prop('onClick');
    const expected = mockFn;
    expect(actual).toEqual(expected);
  });

  test('tag button should have correct data-test attr', () => {
    const actual = wrapper.find('button').prop('data-test');
    const expected = testAttr;
    expect(actual).toEqual(expected);
  });

  test('tag button should have correct children text', () => {
    const actual = wrapper.find('button').prop('children');
    const expected = text;
    expect(actual).toEqual(expected);
  });

  test('tag button should have correct className attr if extensionClass prop was transferred', () => {
    const extClass = 'newClass';
    const props = {
      text,
      name,
      onClick: mockFn,
      'data-test': testAttr,
      extensionClass: extClass,
    };
    const actual = wrapper
      .setProps(props)
      .find('button')
      .prop('className');
    const expected = `button ${extClass}`;
    expect(actual).toEqual(expected);
  });
});
