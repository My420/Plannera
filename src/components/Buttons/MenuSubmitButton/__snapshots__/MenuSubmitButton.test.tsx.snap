// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MenuSubmitButton component should match snapshot 1`] = `
<button
  className="button "
  data-test="test"
  name="testBtn"
  onClick={[MockFunction]}
  type="submit"
>
  Text Button
</button>
`;
